### DISCUSSION TITLE
argo-workflows/running-locally/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9712] [General]
# argo-workflows/running-locally/

https://argoproj.github.io/argo-workflows/running-locally/

**Comment by gcsfred2** [comment]
The text mentions two options in the initial paragraph. It is not clear, however, which sections are for which option.

**Comment by dhaven** [comment]
Hi,
I also found this to be a bit confusing. Basically if you use the dev container you only need to follow the "git clone" step and the "development container" step. You should then have a local k3d cluster + a dev container. You can then exec into the dev container and start the apps using the "make start UI=true" command and all application ports will be forwarded to your localhost

**Comment by ndamclean** [comment]
~I can't find any instructions on how to set up `devcontainer.json` for using dev containers with argo workflows. This seems to be vital information that is missing from this documentation.~

Nevermind, I see the configuration is part of the `argo-workflows` git repository. I just found the documentation a bit confusing (as it seems others did) about which steps apply to dev containers vs installing  components.

> Once installed, go to your argo-workflows folder

I don't think it's entirely clear that "your argo-workflows" folder refers a folder containing the "argo-workflows" git repository. Maybe I was just being a bit dense though :sweat_smile:

**Comment by Joibel** [comment]
> I can't find any instructions on how to set up `devcontainer.json` for using dev containers with argo workflows. This seems to be vital information that is missing from this documentation.

https://argoproj.github.io/argo-workflows/running-locally/#development-container
